frontend/src/api/chatbot.ts

```diff
@@ -28,11 +47,24 @@ export const fetchChatbotReply = async (userMessage: string): Promise<Message> =
 
```

Comment:
fetch doesn’t support timeout natively.  So, wrap it using AbortController to handle cases where the server hangs